app/src/test/java/com/mho/training/features/persondetail/PersonDetailViewModelTest.kt

```diff
@@ -0,0 +1,204 @@
+package com.mho.training.features.persondetail
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.android.domain.Movie
+import com.example.android.domain.Person
+import com.example.android.domain.result.DataResult
+import com.example.android.testshared.mockedMovie
+import com.example.android.testshared.mockedPerson
+import com.example.android.usecases.GetMovieListByPersonUseCase
+import com.example.android.usecases.GetPersonInformationUseCase
+import com.mho.training.utils.Constants
+import com.mho.training.utils.Event
+import com.nhaarman.mockitokotlin2.atLeast
+import com.nhaarman.mockitokotlin2.given
+import com.nhaarman.mockitokotlin2.verify
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import java.io.IOException
+
+/*
+[functionName]Should[Call|Return|Throw|Insert|etc][Function|Object|Exception][WithGiven][When]
+ */
+
+@RunWith(MockitoJUnitRunner::class)
+class PersonDetailViewModelTest {
+
+    @get:Rule
+    val rule: TestRule = InstantTaskExecutorRule()
+
+    @Mock lateinit var getPersonInformationUseCase: GetPersonInformationUseCase
+    @Mock lateinit var getMovieListByPersonUseCase: GetMovieListByPersonUseCase
+
+    @Mock lateinit var observerHasNotMovies: Observer<Boolean>
+    @Mock lateinit var observerHasPersonInformation: Observer<Boolean>
+    @Mock lateinit var observerInfoPerson: Observer<Person>
+    @Mock lateinit var observerLoadingPerson: Observer<Boolean>
+    @Mock lateinit var observerMaxLines: Observer<Int>
+    @Mock lateinit var observerMovies: Observer<List<Movie>>
+    @Mock lateinit var observerShowMoreBiography: Observer<Boolean>
+
+    @Mock lateinit var observerEvents: Observer<Event<PersonDetailViewModel.Navigation>>
+
+    private lateinit var viewModel: PersonDetailViewModel
+
+    @Before
+    fun setUp(){
+        viewModel = PersonDetailViewModel(
+            1,
```

Comment:
***Magic number***: Considere to create a variable whose name describe that `1` number.